torch/cuda/__init__.py

```diff
@@ -352,6 +352,33 @@ def __init__(self, obj):
         super().__init__(idx)
 
 
+
+def _get_device(device: Union[int, str, torch.device] = 'cuda') -> torch.device:
+    r"""Return the torch.device type object from the passed in device.
+
+    Args:
+        device (torch.device or int): selected device.
+    """
+    if isinstance(device, str):
+        device = torch.device(device)
+    elif isinstance(device, int):
+        device = torch.device('cuda', device)
+    return device
+
+
+def _get_generator(device: torch.device) -> torch._C.Generator:
+    r"""Return the CUDA Generator object for the given device.
+
+    Args:
+        device (torch.device): selected device.
+    """
+
+    idx = device.index
+    if idx is None:
+        idx = current_device()
+    return torch.cuda.default_generators[idx]
```

Comment:
wait what if its a CPU device? is this sound? Do you need to assert that this is cuda?